tools/yulPhaser/Population.h

```diff
@@ -77,10 +78,11 @@ class Population
 		std::shared_ptr<FitnessMetric const> _fitnessMetric,
 		std::vector<Chromosome> _chromosomes = {}
 	):
-		Population(
-			std::move(_fitnessMetric),
-			chromosomesToIndividuals(std::move(_chromosomes))
-		) {}
+		Population(std::move(_fitnessMetric), std::vector<Individual>{})
+	{
+		// Can't do it on the initializer list because _fitnessMetric may already be empty due to move()
```

Comment:
It is a shared_ptr, not a unique_ptr - wouldn't it be better to not move instead?

Reply:
`shared_ptr` is a bit heavier than a normal pointer or a number (it does an atomic increment/decrement of its counter on copy) so I generally treat it as something worth moving. This might not be a significant hit if it's not repeated a lot but I find it easier to just always move `shared_ptr`s as a rule rather than think about each individual case.

In this case copying it would be fine but I just wanted to stick to my rule so that later I do not have to stop and think why I'm not moving in this one special case :)

I actually thought moving would be expected for `shared_ptr`. If either way is fine then, well, a copy does make this code shorter. Changed.